ui/components/app/modals/turn-on-metamask-notifications/turn-on-metamask-notifications.tsx

```diff
@@ -0,0 +1,143 @@
+import {
+  selectIsMetamaskNotificationsEnabled,
+  getIsUpdatingMetamaskNotifications,
+} from '../../../../selectors/metamask-notifications/metamask-notifications';
+import { useCreateNotifications } from '../../../../hooks/metamask-notifications/useNotifications';
+import { NOTIFICATIONS_ROUTE } from '../../../../helpers/constants/routes';
+
+import {
+  Box,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  Text,
+  ModalFooter,
+} from '../../../component-library';
+import {
+  AlignItems,
+  BlockSize,
+  BorderRadius,
+  FlexDirection,
+  FontWeight,
+  TextColor,
+} from '../../../../helpers/constants/design-system';
+
+export default function TurnOnMetamaskNotifications() {
+  const { hideModal } = useModalProps();
+  const history = useHistory();
+  const t = useContext(I18nContext);
+  const { listNotifications } = useMetamaskNotificationsContext();
+
+  const isNotificationEnabled = useSelector(
+    selectIsMetamaskNotificationsEnabled,
+  );
+  const isUpdatingMetamaskNotifications = useSelector(
+    getIsUpdatingMetamaskNotifications,
+  );
+
+  const [buttonState, setButtonState] = useState<boolean>(
+    isUpdatingMetamaskNotifications,
+  );
+
+  const { createNotifications, error } = useCreateNotifications();
+
+  const handleTurnOnNotifications = async () => {
+    setButtonState(true);
+    await createNotifications();
+  };
+
+  const handleHideModal = () => {
+    hideModal();
+  };
+
+  useEffect(() => {
+    if (isNotificationEnabled && !error) {
+      history.push(NOTIFICATIONS_ROUTE);
+      hideModal();
+      listNotifications();
+    }
+  }, [isNotificationEnabled, error]);
+
+  const privacyLink = (
+    <Text
+      as="a"
+      href="https://metamask.io/privacy.html"
+      target="_blank"
+      rel="noopener noreferrer"
+      key="privacy-link"
+      color={TextColor.infoDefault}
+    >
+      {t('turnOnMetamaskNotificationsMessagePrivacyLink')}
+    </Text>
+  );
+
+  const strongText = (
+    <Text as="span" fontWeight={FontWeight.Bold} key="strong-text">
+      {t('turnOnMetamaskNotificationsMessagePrivacyBold')}
+    </Text>
+  );
+
+  return (
+    <Modal isOpen onClose={() => handleHideModal()}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader onClose={() => handleHideModal()}>
+          {t('turnOnMetamaskNotifications')}
+        </ModalHeader>
+        <ModalBody>
+          <Box
+            as="img"
+            src="./images/turn-on-metamask-notifications.png"
+            width={BlockSize.Full}
+            borderRadius={BorderRadius.MD}
+            marginBottom={4}
+          />
+          <Text as="p">{t('turnOnMetamaskNotificationsMessageFirst')}</Text>
+          <Text as="p" paddingTop={4}>
+            {
+              // @ts-expect-error: Expected 0-1 arguments, but got an array.
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              t('turnOnMetamaskNotificationsMessageSecond', [privacyLink])
```

Comment:
Fix eslint issue, I don't see why this is needed.

I'm pretty sure the translations should work fine.

Reply:
Use the predefined hook: `useI18nContext`.

This is more loosely (but truthily) typed.

I actually had a PoC a while ago that improved type safety.

Reply:
We have already commented on this.

Reply:
Ticket for me https://consensyssoftware.atlassian.net/browse/NOTIFY-663?atlOrigin=eyJpIjoiNGJiZDEzNzBiNThmNDVkMDhjM2ZlMWRlY2NlYTg2N2YiLCJwIjoiaiJ9